src/jemalloc.c

```diff
@@ -2147,6 +2147,9 @@ imalloc_sample(static_opts_t *sopts, dynamic_opts_t *dopts, tsd_t *tsd,
 		arena_prof_promote(tsd_tsdn(tsd), ret, usize);
 	} else {
 		ret = imalloc_no_sample(sopts, dopts, tsd, usize, usize, ind);
+		if (unlikely(ret == NULL)) {
```

Comment:
Just as a commit title nit, this isn't really a fix so much as a clarification. Falling through is still correct.

Reply:
Right. Just found that `NULL` can also pass `prof_sample_aligned()`.